### PR TITLE
Add bazel support to build nGraph

### PR DESCRIPTION
**Pull request by avijit-nervana**
* Added bazel scripts to be able to build the bridge and nGraph and the INTERPRETER backend
* Steps: 
   * ./configure.sh
   * bazel build libngraph_bridge.so
   * bazel build @ngraph//:libinterpreter_backend.so

Then copy the .so files to the library with rest of the libraries.

### CODE REVIEW

File: bazel/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Build nGraph TensorFlow bridge using bazel
+
+This directory contains scripts necessary to build the nGraph TensorFlow bridge using `bazel`. 
+
+:warning: This is experimental and will change over time. 
+
+## prerequisites
```
Comment (shresthamalik) [inline bazel/README.md]:
```suggestion
## Prerequisites
```

File: examples/tf_cpp_examples/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ The application is linked with TensorFlow C++ library and nGraph-TensorFlow brid
 
 ## prerequisites
 
-The example application requires nGraph-TensorFlow bridge to be built first. Build the ngraph-tf by executing `build_gtf.py` as per the [Option 1] instructions in the main readme. All the files needed to build this example application are located in the ngraph-tf/build directory.
+The example application requires nGraph-TensorFlow bridge to be built first. Build the ngraph-tf by executing `build_gtf.py` as per the [Option 2] instructions in the main readme. All the files needed to build this example application are located in the ngraph-tf/build directory.
```
Comment (shresthamalik) [inline examples/tf_cpp_examples/README.md]:
```suggestion
The example application requires nGraph-TensorFlow bridge to be built first. Build the ngraph-tf by executing `build_ngtf.py` as per the [Option 2] instructions in the main readme. All the files needed to build this example application are located in the ngraph-tf/build directory.
```

**Comment by crlishka** [comment]
TESTNOW

**Comment by crlishka** [comment]
TESTNOW

File: bazel/BUILD [hunk 3]
```diff
@@ -0,0 +1,63 @@
+load("//:cxx_abi_option.bzl", "CXX_ABI")
```
Comment (shresthamalik) [inline bazel/BUILD]: Please add Intel License header

File: bazel/WORKSPACE [hunk 4]
```diff
@@ -0,0 +1,28 @@
+workspace(name = "ngraph_bridge")
```
Comment (shresthamalik) [inline bazel/WORKSPACE]: Intel License Header

File: bazel/ngraph.BUILD [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        ":ngraph_core",
+    ],
+    copts = [
+        "-I external/ngraph/src",
+        "-I external/ngraph/src/ngraph",
+        "-I external/nlohmann_json_lib/include/",
+        '-D SHARED_LIB_PREFIX=\\"lib\\"',
+        '-D SHARED_LIB_SUFFIX=\\".so\\"',
+        '-D NGRAPH_VERSION=\\"0.14.0-rc.1\\"',
+        "-D NGRAPH_DEX_ONLY",
```
Comment (shresthamalik) [inline bazel/ngraph.BUILD]: Please check if we need the SDL flags

File: bazel/BUILD [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        "@ngraph//:ngraph_core",
+    ],
+    copts = [
+        "-pthread", 
+        "-std=c++11", 
+        "-DNDEBUG",
+        '-D SHARED_LIB_PREFIX=\\"lib\\"',
+        '-D SHARED_LIB_SUFFIX=\\".so\\"',
+        "-I logging",
+        "-I external/ngraph/src",
```
Comment (shresthamalik) [inline bazel/BUILD]: Please check if we have SDL flags.

File: bazel/BUILD (hunk 3 above)
Comment (avijit-nervana) [reply bazel/BUILD]: Done

**Pull request merged**